github/teams_discussions_test.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2018 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestTeamsService_ListDiscussions(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/teams/2/discussions", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		testHeader(t, r, "Accept", mediaTypeTeamDiscussionsPreview)
+		testFormValues(t, r, values{
+			"direction": "desc",
+		})
+		fmt.Fprintf(w,
+			`[
+				{
+					"author": {
+						"login": "author",
+						"id": 0,
+						"avatar_url": "https://avatars1.githubusercontent.com/u/0?v=4",
+						"gravatar_id": "",
+						"url": "https://api.github.com/users/author",
+						"html_url": "https://github.com/author",
+						"followers_url": "https://api.github.com/users/author/followers",
+						"following_url": "https://api.github.com/users/author/following{/other_user}",
+						"gists_url": "https://api.github.com/users/author/gists{/gist_id}",
+						"starred_url": "https://api.github.com/users/author/starred{/owner}{/repo}",
+						"subscriptions_url": "https://api.github.com/users/author/subscriptions",
+						"organizations_url": "https://api.github.com/users/author/orgs",
+						"repos_url": "https://api.github.com/users/author/repos",
+						"events_url": "https://api.github.com/users/author/events{/privacy}",
+						"received_events_url": "https://api.github.com/users/author/received_events",
+						"type": "User",
+						"site_admin": false
+					},
+					"body": "test",
+					"body_html": "<p>test</p>",
+					"body_version": "version",
+					"comments_count": 1,
+					"comments_url": "https://api.github.com/teams/2/discussions/3/comments",
+					"created_at": "2018-01-01T00:00:00Z",
+					"last_edited_at": null,
+					"html_url": "https://github.com/orgs/1/teams/2/discussions/3",
+					"node_id": "node",
```

Comment:
I think this is too much boilerplate - the practice in other portions of the API is to only spell out a minimal response in unit tests.  Example: https://github.com/google/go-github/blob/12126fbfe000a09047ebdeb2b5160be3af88ab9e/github/pulls_test.go#L225-L228 Mainly for consistency, I would recommend only specifying a minimal few fields to get working tests. Also, writing out every field can increase the cost of maintenance. Does that make sense to you?

Reply:
Shouldn't newly added struct be tested for its ability to parse a full response? I made sure to do it only once, and that is why Get* tests only have minimal fields. I can remove Author field as that should be covered by users_test.